src/main.dart

```diff
@@ -1,8 +1,107 @@
-// TODO: Run task list app
+
+import 'dart:io';
+import 'task_list.dart';
+
+ TaskList tasks = TaskList();
 void main(List<String> args) {
-  print('Hello World');
+  printMenu();
+
+  bool choiceValid = false;
+  int choice = -1;
+  while(!choiceValid) {
+    try {
+      stdout.write("Enter your choice: ");
+      choice = int.parse(stdin.readLineSync()!);
+      choiceValid = true;
+  } catch(e) {
+    print("Please enter a number");
+  }
+}
+switch(choice) {
+  case 1:
+  addTask();
+  break;
+  case 2:
+  editTask();
+  break;
+  case 3:
+  deleteTask();
+  break;
+  case 4:
+  markTaskAsCompleted();
+  break;
+  case 5:
+  tasks.printTasks();
+  break;
+  case 6:
+  print("Thank you for using the Task List App");
+  default:
+  print("Invalid choice");
+  break;
+
+
+}
+}
+void printMenu() {
+  print("==========================================");
+  print("Task list app");
+  print("==========================================");
+  print("Select your service: ");
+  print('1. Add a task');
+  print('2. Edit a task');
+  print('3. Delete a task');
+  print('4. Mark a task as completed');
+  print('5. Print tasks');
+  print('6. Exit');
+  print("==========================================");
 }
 
-class TaskListApp {
-  // TODO: Implement Task list app
+void addTask() {
+  stdout.write("Enter task title: ");
+  String? title = stdin.readLineSync() ??"";
+  if(title.trim() == ""){
+    print("Title cannot be empty");
+    return;
+  }
+  stdout.write("Enter task description: ");
+  String? description = stdin.readLineSync() ??"";
+  if(description.trim() == ""){
+    print("Description cannot be empty");
+    return;
+  }
+  tasks.addTask(title, description);
 }
+
+void editTask() {
+  stdout.write("Enter the index of the task to edit: ");
+  int index = int.parse(stdin.readLineSync()!);
+  if(index < 0 || index >= tasks.length) {
+    print("Invalid task index.");
+    return;
+  }
+  stdout.write("Enter the new task title: ");
+  String newTitle = stdin.readLineSync()!;
+  tasks.editTask(index, newTitle);
```

Comment:
should try searching by ID or by content instead of just by index. use [firstWhere](https://api.dart.dev/stable/3.2.4/dart-core/Iterable/firstWhere.html) function